### PR TITLE
Add dependency auditing to pipelines

### PR DESCRIPTION
**Pull request by svenaas**
Part of #458 

## Changes proposed in this pull request:
- Add [pip-audit](https://github.com/pypa/pip-audit) as a dev dependency.
- Add `pip-audit.sh` script and new `audit-dependencies` tasks which run it in the CI pipelines.

## security considerations
None. This introduces a new check to help ensure that out-of-date dependencies get resolved.

### CODE REVIEW

File: ci/tasks/pip-audit.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+pip install -r requirements-dev.txt
```
Comment (drewbo) [inline ci/tasks/pip-audit.sh:5]: For the audit you can install just `pip-audit` to speed this up a bit

**Comment by svenaas** [comment]
> Great start! Can you add the audit task to the staging and prod pipelines?

I've added it to `pipeline.yml` now; I'm not sure whether it would be wise to adjust any of its details there.

**Comment by svenaas** [comment]
> @svenaas for staging/prod (which both run off `pipeline.yml`) I would set the audit to run off the git resource rather than the pr resource (`src-((deploy-env))`). Right now we have a little duplication in triggers here. Eventually we can match the pages-core pattern but that will take a bit of work

@drewbo How's this latest commit look? I changed the notification text a little as well.

File: ci/tasks/pip-audit.sh (hunk 1 above)
Comment (svenaas) [reply ci/tasks/pip-audit.sh]: Done. Still works and it's definitely lighter.

**Review by drewbo — APPROVED** [review]

**Pull request merged**